### PR TITLE
Add filtering options to piers and berths

### PR DESCRIPTION
**Pull request by ahie**
Enables filtering berths by harbor and pier, and piers by harbor. This enables simplifying deeply nested queries on the FE.

### CODE REVIEW

**Review by a-rmz — COMMENTED** [review]
Could you add some tests for this? 😇

File: resources/schema/types.py [hunk 1]
```diff
@@ -83,6 +85,14 @@ class BerthNodeFilterSet(django_filters.FilterSet):
...
     )
     is_available = django_filters.BooleanFilter()
+    harbor = GlobalIDFilter(method="filter_by_harbor")
+    pier = GlobalIDFilter(method="filter_by_pier")
+
+    def filter_by_pier(self, queryset, name, value):
+        return queryset.filter(pier__id=from_global_id(value))
+
+    def filter_by_harbor(self, queryset, name, value):
+        return queryset.filter(pier__harbor__id=from_global_id(value))
```
Comment (a-rmz) [inline resources/schema/types.py:95]:
```suggestion
    harbor = GlobalIDFilter(field_name="pier__harbor", lookup_expr="id")
    pier = GlobalIDFilter(field_name="pier", lookup_expr="id")
```

**Comment by a-rmz** [comment]
One last question: what happens if you pass both `pier/harbor` filters when the pier belongs to a different pier?

File: resources/schema/types.py (hunk 1 above)
Comment (ahie) [reply resources/schema/types.py]: Done.

**Comment by ahie** [comment]
> Could you add some tests for this? 😇

Done.

> One last question: what happens if you pass both pier/harbor filters when the pier belongs to a different pier?

It will just return zero results. Added test case for this.

**Review by a-rmz — APPROVED** [review]
Looks good now 👍 one last thing: could you add ass comment to the `BerthNode` description that if you use both `harbor` and `pier` filters with a pier that does not belong the harbour, it will not return anything.

**Comment by ahie** [comment]
> Looks good now 👍 one last thing: could you add ass comment to the `BerthNode` description that if you use both `harbor` and `pier` filters with a pier that does not belong the harbour, it will not return anything.

Added description to the `berths` query.

**Review by a-rmz — APPROVED** [review]

**Pull request merged**